testsrc/main/mondrian/xmla/XmlaDimensionPropertiesTest.java

```diff
@@ -0,0 +1,824 @@
+/*
+// This software is subject to the terms of the Eclipse Public License v1.0
+// Agreement, available at the following URL:
+// http://www.eclipse.org/legal/epl-v10.html.
+// You must accept the terms of that agreement to use this software.
+//
+// Copyright (C) 2005-2015 Pentaho and others
+// All Rights Reserved.
+*/
+package mondrian.xmla;
+
+import mondrian.test.DiffRepository;
+import mondrian.test.TestContext;
+
+import java.util.Properties;
+
+/**
+ * Test one axis's dimensions properties is added to it own hierarchy
+ *  - fix of MONDRIAN-2302 issue .
+ *
+ * @author Yury_Bakhmutski.
+ */
+public class XmlaDimensionPropertiesTest extends XmlaBaseTestCase {
+
+    public void testOneHierarchyProperties() throws Exception {
+        final String schema = defineSchema();
+        TestContext context = TestContext.instance().withSchema(schema);
+        String requestType = "EXECUTE";
+        Properties props = getDefaultRequestProperties(requestType);
+        doTest(requestType, props, context);
+    }
+
+    public void testTwoHierarchiesProperties() throws Exception {
+        final String schema = defineSchema();
+        TestContext context = TestContext.instance().withSchema(schema);
+        String requestType = "EXECUTE";
+        Properties props = getDefaultRequestProperties(requestType);
+        doTest(requestType, props, context);
+    }
+
+    private String defineSchema() {
```

Comment:
We don't copy the whole schema in tests. There are plenty of helper methods to help you make any modifications that are necessary. See TestContext.createSubstitutingCube and others.